web/Web/web/CAL/views.py

```diff
@@ -198,3 +199,70 @@ def get_ajax(self, request, *args, **kwargs):
         except CALError as e:
             # TODO: add proper http response for CAL errors
             return JsonResponse({"message": "Ops! CALError."}, status=404)
+
+
+class DSLoggingView(views.CsrfExemptMixin,
+                    views.LoginRequiredMixin,
+                    RetrievalMethodPermissionMixin,
+                    views.JsonRequestResponseMixin,
+                    generic.View):
+    """
+    View to log info of Dynamic Sampling
+    """
+    require_json = False
+
+    def post(self, request, *args, **kwargs):
+        session = self.request.user.current_session.uuid
+
+        try:
+            stratum_num = self.request_json.get(u"stratum_number")
+            stratum_size = self.request_json.get(u"stratum_size")
+            T = self.request_json.get(u"T")
+            N = self.request_json.get(u"N")
+            R = self.request_json.get(u"R")
+            current_sample_size = self.request_json.get(u"n")
+
+            # Get docs in current stratum
+            docs = CALFunctions.get_stratum_documents(str(session))
+            # Get sampled docs
+            sampled_docs, _ = CALFunctions.get_documents(str(session), -1)
+
+        except KeyError:
+            error_dict = {u"message": u"Errors when getting DS info."}
+            return self.render_bad_request_response(error_dict)
+
+        if not docs:
+            return self.render_bad_request_response({u"message": u"Empty set for stratum {}".format(stratum_num)})
+
+        if not sampled_docs:
+            return self.render_bad_request_response({u"message": u"Empty sample for stratum {}".format(stratum_num)})
+
+        if int(stratum_num) == 1:
+            current_sample_size = 1
```

Comment:
Is there a reason we need to hard code the `current_sample_size`?

Reply:
Because if we use seed document for the first stratum, the first `stratum` in vector `stratums` in S-CAL is empty, meaning that the returned `current_sample_size` is 0. So Amir and I decided to hardcode the first sample size to be 1.